packages/operator/pkg/apiserver/routes/v1/deployment/modelroute_validation.go

```diff
@@ -81,7 +82,7 @@ func (mrv *MrValidator) validateMainParameters(mr *deployment.ModelRoute) (err e
 		}
 	}
 	if mr.Spec.Mirror != nil && len(*mr.Spec.Mirror) != 0 {
-		if _, k8sError := mrv.mdRepository.GetModelDeployment(*mr.Spec.Mirror); k8sError != nil {
+		if _, k8sError := mrv.mdService.GetModelDeployment( context.TODO(), *mr.Spec.Mirror); k8sError != nil {
```

Comment:
I think, as the next iteration of this refactoring, the validation itself should be moved from controller to service. Service entity should fully reflect the service behaviour, while controller just binds HTTP terms (URL, query params, body) to this logic. Validation is of course a part of logic, therefore service.
I can do that myself when I have a spare moment if you want.

Reply:
Yes, I agree with u.

Reply:
But let's do that in different PR.

Reply:
Ok, sure. I'll create a Tech-debt task :)